docs/proposals/nap-waf.md

```diff
@@ -0,0 +1,753 @@
+
+- Reference compiled logging profile bundles from remote sources
+- Same fetch and validation mechanisms as policy bundles
+- Support for checksums and retry policies
+
+**Destination Types:**
+
+- `stderr`: Output to container stderr
+- `file`: Write to specified file path (must be mounted to host for waf-enforcer container access)
+- `syslog`: Send to syslog server via TCP (recommend local proxy for secure forwarding)
+
+**Generated NGINX Configuration Examples:**
+
+```nginx
+# Built-in profile to stderr
+app_protect_security_log log_all stderr;
+
+# Custom profile to file
+app_protect_security_log /shared_volume/custom-log-profile.tgz /var/log/app_protect/security.log;
+
+# Built-in profile to syslog
+app_protect_security_log log_blocked syslog:server=syslog-svc.default:514;
+```
+
+### Policy Fetch Failure Handling
+
+In the cases where there are policy fetch failures:
+
+**First-Time Policy Fetch Failure:**
+
+- Route configuration is **not applied** - no WAF protection enabled
+- Route remains unprotected until policy becomes available
+
+**Policy Update Failure:**
+
+- **Existing policy remains in effect** - no disruption to current protection
+- WAF protection continues with the last successfully deployed policy
+
+**Retry Behavior:**
+
+- Configurable retry policy with exponential backoff
+- No service disruption during retry attempts
+- Detailed error messages for troubleshooting
+
+### NGF Integration Architecture
+
+The integration leverages NGF's existing architecture:
+
+- **Single NGF Pod**: Centralized control plane in `nginx-gateway` namespace manages all WAF operations
+- **Per-Gateway Deployment**: Each Gateway with WAF enabled gets a dedicated multi-container NGINX Pod
+- **Selective WAF Enablement**: Only Gateways configured with WAF-enabled NginxProxy resources deploy NAP v5 containers
+- **Centralized Policy Management**: NGF controllers fetch policies and distribute them to appropriate NGINX Pods via the existing Agent gRPC connection
+
+## API, Customer Driven Interfaces, and User Experience
+
+### NginxProxy Resource Extension
+
+Users enable WAF through the NginxProxy resource,
+
+```yaml
+apiVersion: gateway.nginx.org/v1alpha2
+kind: NginxProxy
+metadata:
+  name: nginx-proxy-waf
+  namespace: nginx-gateway
+spec:
+  # Simple WAF enablement (hides multi-container complexity)
+  waf:
+    enabled: true
+    # configuration tweaks optional, e.g.:
+#   kubernetes:
+#     deployment:
+#       # NGINX container with NAP module (will set to default if waf enabled is true but these values are not configured)
+#       container:
+#         image:
+#           repository: private-registry.nginx.com/nginx-gateway-fabric/nginx-plus-waf
+#           tag: "2.1.0"
+
+#       # NAP v5 required containers (will set to defaults if waf enabled is true but these values are not configured)
+#       wafContainers:
+#         enforcer:
+#           image:
+#             repository: private-registry.nginx.com/nap/waf-enforcer
```

Comment:
These images already exist? No need for us to build/ship any additional images than we already do?